### PR TITLE
create images from csv data

### PR DESCRIPTION
**Pull request by blanche**
see #2

### CODE REVIEW

File: src/data2img.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import os
+
+IMG_SIZE = 48
+csv_file_name = '../data/head.csv'
+data_output_dir = '../data/img'
+
+
+# based on https://stackoverflow.com/questions/312443/how-do-you-split-a-list-into-evenly-sized-chunks
+# based on https://stackoverflow.com/questions/434583/what-is-the-fastest-way-to-draw-an-image-from-discrete-pixel-values-in-python
+def csv2img(file_name):
```
Comment (blanche) [inline src/data2img.py]: this should become some util method and moved somewhere else, as we are going to need it in some places #7

File: src/data2img.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+if __name__ == "__main__":
+    # convert image to data
+    # returns data in format [(expression, numpy array)]
+    img_data = csv2array(csv_file_name)
+
+    if not os.path.exists(data_output_dir):
+        os.makedirs(data_output_dir)
+    # save first 10 images for demo purpose
+    for k in range(0, 9):
```
Comment (utkuozbulak) [inline src/data2img.py]: Instead of saving first 10 images can you edit a bit to save first few samples of each emotion(7) ? ( sad, angry etc...)

**Comment by utkuozbulak** [comment]
I think it would be cool to use few images at main readme file, can you add couple of images to PR as well ?

**Review by arahayrabedian — DISMISSED** [review]
works! there's a few external dependencies required, i've made a note of the crap i installed since i had never set up scipy and numpy in a virtualenv before, i'll add those in once the two PRs are in.

**Comment by andreaseliasson** [comment]
Works (after installing PIL/Pillow) [scipy.misc](https://docs.scipy.org/doc/scipy/reference/misc.html)

File: src/data2img.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
+import numpy as np
+import scipy.misc
+import csv
+import os
+
+IMG_SIZE = 48
+csv_file_name = '../data/head.csv'
```
Comment (andreaseliasson) [inline src/data2img.py]: Add to .gitignore

**Review by andreaseliasson — APPROVED** [review]

File: src/data2img.py (hunk 3 above)
Comment (arahayrabedian) [reply src/data2img.py]: also, between this branch and @utkuozbulak 's we have two different csv names, i prefer the one that unrar's straight out of the file downloaded: `fer2013.csv` as in the other branch

File: src/data2img.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+# based on https://stackoverflow.com/questions/312443/how-do-you-split-a-list-into-evenly-sized-chunks
+# based on https://stackoverflow.com/questions/434583/what-is-the-fastest-way-to-draw-an-image-from-discrete-pixel-values-in-python
+def csv2array(file_name):
+    img_list = []
+    csv_file = csv.DictReader(open(file_name))
+
+    for csv_row in csv_file:
+        data = np.zeros((IMG_SIZE, IMG_SIZE), dtype=np.uint8)
+        pixel_data = csv_row['pixels'].split()
+        for i in range(0, IMG_SIZE - 1):
```
Comment (arahayrabedian) [inline src/data2img.py]: by the by - @utkuozbulak noticed that there are black lines at the bottom of the output, removing the `- 1` fixes it

**Review by arahayrabedian — DISMISSED** [review]
black bars need to be gone

File: src/data2img.py (hunk 3 above)
Comment (blanche) [reply src/data2img.py]: yea it should be fer2013.csv - i didnt want to work with the 200mb file here though and created a `head fer2013.csv > head.csv` of it - i see this more of a utility method to be called somewhere else #7 , i guess we are not going to run this file

File: src/data2img.py (hunk 4 above)
Comment (blanche) [reply src/data2img.py]: fixed

**Pull request merged**